### DISCUSSION TITLE
The entrify commands should prompt for the entry type name and handle

### DISCUSSION THREAD

**Discussion by MoritzLost** [craftcms/cms#12894] [Ideas] [answered]
The `entrify/categories` and `entrify/tags` commands prompt for the name and handle for the new section they create to replace the existing category or tag group. However, they _don't_ prompt for the entry type's name or handle, instead they always use `Default` and `default`. This is a bit inconvenient – I don't like having all my entry types named `default`, because it makes them borderline unusable for entry queries. So I have to manually change the entry type's name and handle after the running the commands. However, this also means I have to adjust the `entrify` commands to be executed in the production environment, because those include a parameter `--entry-type=default`, which will not work after I've changed the entry type's handle.

It would be great if the `entrify` commands could just prompt for the entry type name and handle instead of using those default values!

**Comment by brandonkelly** [comment, marked as answer]
I think it makes sense to just assume `default` for now, to be consistent with the initial entry type created when you create a new section.

We will _need_ to change this in Craft 5 when we decouple entry types from sections, though, so I’ll make a note to update this when that happens.

**Reply by MoritzLost** [reply]
@brandonkelly Hm, I understand, though I don't agree. I don't really understand why entry types are always named `default` on creation – does that ever make sense? This will only lead to having multiple entry types named `default` in different sections, meaning you can't really use the entry type in entry queries, unless you also specify the section. And regarding the decoupling in Craft 5 – how do you plan to handle the case that multiple entry types in different sections have the same handle `default`? I assume this won't be possible any more. In this case, wouldn't it make sense to use a different default now to make the transition to Craft 5 easier in the future?

Would it very complicated to have the command prompt for the entry type name and handle same as it does for the section? It could still suggest `Default` and `default`, but allowing the user to change those values during the command would certainly make the entrification easier. And it would avoid potential errors if the user changes the entry type's handle but forgets to change it inside the command for the production environment.

**Reply by brandonkelly** [reply]
> I don't really understand why entry types are always named default on creation – does that ever make sense?

See #7078 for the discussion that lead to that.

> Would it very complicated to have the command prompt for the entry type name and handle same as it does for the section? It could still suggest Default and default, but allowing the user to change those values during the command would certainly make the entrification easier. 

Fair enough. Just released 4.4.3 with new prompts for the initial entry type’s name and handle. (f127649719c730e6517bb06ab3d261d311e8ff70)

**Reply by MoritzLost** [reply]
@brandonkelly Awesome, thanks! 🚀
